lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1196,7 +1196,7 @@ namespace Js
                     }
                 }
 
-                SetAttributesHelper(instance, propertyId, propertyIndex, attributes, (ObjectSlotAttributes)(attr | ObjectSlotAttr_Accessor));
+                SetAttributesHelper(instance, propertyId, propertyIndex, attributes, (ObjectSlotAttributes)((attr & ~ObjectSlotAttr_Deleted) | ObjectSlotAttr_Accessor));
```

Comment:
Can this just be:

```cpp
attr ^ ObjectSlotAttr_Deleted | ObjectSlotAttr_Accessor
```

?

Also, any idea why the already-merge commits are showing up in this PR? Maybe rebase?

Reply:
The real problem here seems to be that we were able to create a property on a PathTypeHandler with the deleted attribute. That shouldn't be possible. Can you see how it happened?

Reply:
The object has SimpleTypeHandler<3> type handler (which has proto, length and name). Once we delete the length, we mark the attribute for that slot as deleted.
Around the time when we are about to add the accessor - we converted that handler to pathtypehandlerwithattr with the deleted property attribute for the 'length'.
At the time of adding the accessor - we didn't remove that attribute.

Reply:
```
    template<size_t size>
    BOOL SimpleTypeHandler<size>::SetAccessors(DynamicObject* instance, PropertyId propertyId, Var getter, Var setter, PropertyOperationFlags flags)
    {
        if (DoConvertToPathType(instance->GetDynamicType()))
        {
            return ConvertToPathType(instance)->SetAccessors(instance, propertyId, getter, setter, flags);
        }
        else
        {
            // Don't attempt to share cross-site function types
            return ConvertToDictionaryType(instance)->SetAccessors(instance, propertyId, getter, setter, flags);
        }
    }
```

Reply:
I'm saying that setting a property on a path type handler with the deleted attribute shouldn't work, since path type handlers don't support deleted properties. It should have forced conversion to some sort of dictionary type handler. If there's a simple type with a deleted property, and we convert to path type without re-adding the deleted property, we'll have trouble even with the fix you're proposing.

Reply:
It seems the Pathtypehandler supports deleted property. In many of the function in pathtypehandler.cpp you can see there are checks for ``` attr & ObjectSlotAttr_Deleted```
you can find that in ```HasProperty/SetProperty/GetProperty/DeleteProperty functions```.

Reply:
It definitely doesn't. Those lines are there to prepare for support, but it's not there.

Reply:
huh, that is strange.
I can try looking into converting to DictionaryType at the time we are adding an accessor to simple type. Do you know if we marked something on the simple type that any property got deleted? ```DoConvertToPathType``` does not have that information.

Reply:
If simple type does support deleted properties, then "deleted" must be recorded in the property's attributes.